markdownhighlighter.cpp

```diff
@@ -1749,7 +1748,7 @@ void MarkdownHighlighter::highlightCheckbox(const QString &text, int curPos)
     const bool hasXorSpace = midChar == QLatin1Char(' ') || midChar == QLatin1Char('x') || midChar == QLatin1Char('X');
     const bool hasDash = midChar == QLatin1Char('-');
 
-    if (hasOpeningBracket && hasClosingBracked && (hasXorSpace | hasDash)) {
+    if (hasOpeningBracket && hasClosingBracked && (hasXorSpace || hasDash)) {
```

Comment:
Spelling of hasClosingBracked can also be fixed if you want to

Reply:
Good idea